lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -239,6 +242,7 @@ CDataFrameTrainBoostedTreeClassifierRunner::inferenceModelDefinition(
 }
 
 // clang-format off
+const std::size_t CDataFrameTrainBoostedTreeClassifierRunner::MAX_NUMBER_CLASSES{30};
```

Comment:
It's probably worth adding a comment to say that if the `30` is ever changed it also needs changing in https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java#L57

Reply:
This likely will change at least before 7.7. I'll add a comment.

Reply:
I linked the named variable: since this comment is unlikely to get changed if the code is refactored or moved, but hopefully we won't change the name. See [1ce554b](https://github.com/elastic/ml-cpp/pull/1056/commits/1ce554bd5baeb18b97fc119335418d63ea3f96e4).